src/shared/utils/extractUploads.js

```diff
@@ -1,86 +0,0 @@
-import countBy from 'lodash/countBy'
-import groupBy from 'lodash/groupBy'
-
-import { UploadStateEnum, UploadTypeEnum } from 'shared/utils/commit'
-
-function humanReadableOverview(state, count) {
-  if (state === UploadStateEnum.error) return 'errored'
-  if (state === UploadStateEnum.processed) return 'successful'
-  if (state === UploadStateEnum.complete) return 'carried forward'
-  if (state === UploadStateEnum.started) return 'started'
-}
-
-function deleteDuplicateCFFUploads({ uploads }) {
-  const nonCFFFlags = []
-  const duplicateUploads = uploads && [...uploads]
-
-  // Get all the flags from "uploaded" uploads
-  duplicateUploads?.forEach((upload) => {
-    if (
-      upload?.uploadType !== UploadTypeEnum.CARRIED_FORWARD &&
-      upload?.flags
-    ) {
-      nonCFFFlags.push(...upload.flags)
-    }
-  })
-
-  // Filter out cff uploads with repeated flags
-  // We're looping in reverse as splicing reindexes the array, making us
-  // skip the index that's removed. Indices are preserved when you
-  // loop backwards.
-  for (let index = duplicateUploads.length - 1; index >= 0; index--) {
-    const upload = duplicateUploads[index]
-    if (
-      upload?.uploadType === UploadTypeEnum.CARRIED_FORWARD &&
-      upload?.flags
-    ) {
-      for (const flag of upload.flags) {
-        if (nonCFFFlags.includes(flag)) {
-          duplicateUploads.splice(index, 1)
-          break
-        }
-      }
-    }
-  }
-
-  return duplicateUploads
-}
-
-const createUploadOverview = ({ uploads }) =>
-  Object.entries(countBy(uploads, (upload) => upload.state))
```

Comment:
Here we're doing another full pass through all the uploads to generate the "counts" for each state

Then another pass to do the groupBy and filter

Then another pass to do _another_ group by on providers

effectively 3 passes through the uploads to do a couple minor operations. In the new implementation we combine it all into a single pass